client/functions.lua

```diff
@@ -918,3 +946,65 @@ function QBCore.Functions.StartParticleOnEntity(dict, ptName, looped, entity, bo
     end
     return particleHandle
 end
+
+function QBCore.Functions.GetStreetNametAtCoords(coords)
+    local streetname1, streetname2  	= GetStreetNameAtCoord(coords.x, coords.y, coords.z)
+    return { main = GetStreetNameFromHashKey(streetname1), cross = GetStreetNameFromHashKey(streetname2) }
+end
+
+function QBCore.Functions.GetZoneAtCoords(coords)
+    return GetLabelText(GetNameOfZone(coords))
+end
+
+function QBCore.Functions.GetCardinalDirection(entity)
+    entity = DoesEntityExist(entity) and entity or PlayerPedId()
+    if DoesEntityExist(entity) then
+        local heading = GetEntityHeading(entity)
+        if ((heading >= 0 and heading < 45) or (heading >= 315 and heading < 360)) then
+            return "North"
+        elseif (heading >= 45 and heading < 135) then
+            return "West"
+        elseif (heading >= 135 and heading < 225) then
+            return "South"
+        elseif (heading >= 225 and heading < 315) then
+            return "East"
+        end
+    else
+        return "Cardinal Direction Error"
+    end
+end
+
+function QBCore.Functions.GetCurrentTime()
+    local hour      = GetClockHours()
+    local minute    = GetClockMinutes()
+    local obj       = {}
+
+    if hour <= 12 then
+        obj.ampm = "AM"
+    elseif hour >= 13 then
+        obj.ampm = "PM"
+        hour = hour - 12
+    end
+
+    if minute <= 9 then
+        minute = "0" .. minute
+    end
+
+    obj.hour    = hour
+    obj.minute  = minute
+
+    return obj
```

Comment:
Return the unformatted time here as well for the convenience of getting AM/PM numbers and just numbers

Reply:
Should now be complete with both AM/PM hours and also 24-hour clock formatted. However, the linting report is showing an error for some reason?

Reply:
I don't know what the error is, so I'll ask the maintainer of the linter, I'll get back to you

Reply:
I got the result back, the errors are because you added `local` in front of a table value that is already local, so `local obj.min` needs to be just `obj.min` and the same for the hour

Reply:
🤦🏼‍♂️ and this is why you should check the changes in the game before editing. Should be fixed now, mate.

Would you like me to close this PR and then submit as an updated one to keep the repo clean?

Reply:
Yes please